### ISSUE TITLE
Question: how does the "next" in chainKeyBindingCommands works

### ISSUE THREAD

**Issue by zlv-thisF** [remirror/remirror#279]
remirror's code organization based on prosemirror is  amazing !!!!<br>
I check the source code of extension-manager and really interested in the **keymaps** method. it has a **chainKeyBindingCommands** method to loop all the commands from **keys** method in each extension with it's priority.  <br>
the question is that i don not see any extension's keys methods call the **next** manually., in fact **convertCommand** not inject next parameter at all... it confused me.... how does next augument work?
```javascript
export const convertCommand = <GSchema extends EditorSchema = any, GExtraParams extends object = {}>(
  commandFunction: ProsemirrorCommandFunction<GSchema>,
): CommandFunction<GSchema, GExtraParams> => ({ state, dispatch, view }) =>
  commandFunction(state, dispatch, view);
```

```javascript
export const chainKeyBindingCommands = (
  ...commands: KeyBindingCommandFunction[]
): KeyBindingCommandFunction => params => {
  if (!commands.length) {
    return false;
  }

  const [command, ...rest] = commands;

  let calledNext = false;

  const createNext = (...nextCommands: KeyBindingCommandFunction[]): (() => boolean) => () => {
    if (!nextCommands.length) {
      return false;
    }
    calledNext = true;
    const [, ...nextRest] = nextCommands;
    return chainKeyBindingCommands(...nextCommands)({ ...params, next: createNext(...nextRest) });
  };
  const next = createNext(...rest);
  const status = command({ ...params, next });
  if (calledNext || status) {
    return status;
  }
  return next();
};
```
thanks!

**Comment by ifiokjr** [comment]
> remirror's code organization based on prosemirror is amazing !!!!

Thanks for the kind words @zlv-thisF 

### `next`

The `next` parameter allows a keybinding to run its own commands when a key press happens while at the same time running all the other matching keybindings from the editor. 

Calling `next` says, "remirror, please run every matching keybinding provided by the lower priority extensions". If anything happened it will return true, and if nothing happened it will return false.


There are some inline docs available as well or you can read the PR #234 for more details.

https://github.com/remirror/remirror/blob/aa0fe5736e6da7cd1b3bb20223a4ee299ea5d1d7/%40remirror/core-types/src/core-types.ts#L124-L137


### `convertCommand`

`convertCommand` converts the remirror command (with `state`, `dispatch` and `view` as the first parameter) into a `Prosemirror` command (with `state`, `dispatch` and `view` as separate arguments).

**Comment by zlv-thisF** [comment]
COOOOOL!!!
